# ncurses 6.0 - patch 20160702 - Thomas E. Dickey
#
# ------------------------------------------------------------------------------
#
# Ncurses 6.0 is at
# 	ftp.gnu.org:/pub/gnu
#
# Patches for ncurses 6.0 can be found at
# 	ftp://invisible-island.net/ncurses/6.0
#	http://invisible-mirror.net/archives/ncurses/6.0 
#
# ------------------------------------------------------------------------------
# ftp://invisible-island.net/ncurses/6.0/ncurses-6.0-20160702.patch.gz
# patch by Thomas E. Dickey <dickey@invisible-island.net>
# created  Sun Jul  3 00:26:06 UTC 2016
# ------------------------------------------------------------------------------
# NEWS                             |    5 ++++-
# VERSION                          |    2 +-
# dist.mk                          |    4 ++--
# package/debian-mingw/changelog   |    4 ++--
# package/debian-mingw64/changelog |    4 ++--
# package/debian/changelog         |    4 ++--
# package/mingw-ncurses.nsi        |    4 ++--
# package/mingw-ncurses.spec       |    2 +-
# package/ncurses.spec             |    2 +-
# test/list_keys.c                 |   26 +++++++++++++++++++-------
# 10 files changed, 36 insertions(+), 21 deletions(-)
# ------------------------------------------------------------------------------
Index: NEWS
Prereq:  1.2624 
--- ncurses-6.0-20160625+/NEWS	2016-06-25 21:23:38.000000000 +0000
+++ ncurses-6.0-20160702/NEWS	2016-07-02 18:29:57.000000000 +0000
@@ -25,7 +25,7 @@
 -- sale, use or other dealings in this Software without prior written        --
 -- authorization.                                                            --
 -------------------------------------------------------------------------------
--- $Id: NEWS,v 1.2624 2016/06/25 21:23:38 tom Exp $
+-- $Id: NEWS,v 1.2626 2016/07/02 18:29:57 tom Exp $
 -------------------------------------------------------------------------------
 
 This is a log of changes that ncurses has gone through since Zeyd started
@@ -45,6 +45,9 @@
 Changes through 1.9.9e did not credit all contributions;
 it is not possible to add this information.
 
+20160702
+	+ improve test/list_keys.c, using $TERM if no parameters areg given.
+
 20160625
 	+ build-fixes for ncurses "test_progs" rule.
 	+ amend change to CF_CC_ENV_FLAGS in 20160521 to make multilib build
Index: VERSION
--- ncurses-6.0-20160625+/VERSION	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/VERSION	2016-07-02 17:00:31.000000000 +0000
@@ -1 +1 @@
-5:0:9	6.0	20160625
+5:0:9	6.0	20160702
Index: dist.mk
Prereq:  1.1112 
--- ncurses-6.0-20160625+/dist.mk	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/dist.mk	2016-07-02 17:00:31.000000000 +0000
@@ -25,7 +25,7 @@
 # use or other dealings in this Software without prior written               #
 # authorization.                                                             #
 ##############################################################################
-# $Id: dist.mk,v 1.1112 2016/06/25 17:17:16 tom Exp $
+# $Id: dist.mk,v 1.1113 2016/07/02 17:00:31 tom Exp $
 # Makefile for creating ncurses distributions.
 #
 # This only needs to be used directly as a makefile by developers, but
@@ -37,7 +37,7 @@
 # These define the major/minor/patch versions of ncurses.
 NCURSES_MAJOR = 6
 NCURSES_MINOR = 0
-NCURSES_PATCH = 20160625
+NCURSES_PATCH = 20160702
 
 # We don't append the patch to the version, since this only applies to releases
 VERSION = $(NCURSES_MAJOR).$(NCURSES_MINOR)
Index: package/debian-mingw/changelog
--- ncurses-6.0-20160625+/package/debian-mingw/changelog	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/package/debian-mingw/changelog	2016-07-02 17:00:31.000000000 +0000
@@ -1,8 +1,8 @@
-ncurses6 (6.0+20160625) unstable; urgency=low
+ncurses6 (6.0+20160702) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 25 Jun 2016 13:17:16 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 02 Jul 2016 13:00:31 -0400
 
 ncurses6 (5.9-20131005) unstable; urgency=low
 
Index: package/debian-mingw64/changelog
--- ncurses-6.0-20160625+/package/debian-mingw64/changelog	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/package/debian-mingw64/changelog	2016-07-02 17:00:31.000000000 +0000
@@ -1,8 +1,8 @@
-ncurses6 (6.0+20160625) unstable; urgency=low
+ncurses6 (6.0+20160702) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 25 Jun 2016 13:17:16 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 02 Jul 2016 13:00:31 -0400
 
 ncurses6 (5.9-20131005) unstable; urgency=low
 
Index: package/debian/changelog
--- ncurses-6.0-20160625+/package/debian/changelog	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/package/debian/changelog	2016-07-02 17:00:31.000000000 +0000
@@ -1,8 +1,8 @@
-ncurses6 (6.0+20160625) unstable; urgency=low
+ncurses6 (6.0+20160702) unstable; urgency=low
 
   * latest weekly patch
 
- -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 25 Jun 2016 13:17:16 -0400
+ -- Thomas E. Dickey <dickey@invisible-island.net>  Sat, 02 Jul 2016 13:00:31 -0400
 
 ncurses6 (5.9-20120608) unstable; urgency=low
 
Index: package/mingw-ncurses.nsi
Prereq:  1.164 
--- ncurses-6.0-20160625+/package/mingw-ncurses.nsi	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/package/mingw-ncurses.nsi	2016-07-02 17:00:31.000000000 +0000
@@ -1,4 +1,4 @@
-; $Id: mingw-ncurses.nsi,v 1.164 2016/06/25 17:17:16 tom Exp $
+; $Id: mingw-ncurses.nsi,v 1.165 2016/07/02 17:00:31 tom Exp $
 
 ; TODO add examples
 ; TODO bump ABI to 6
@@ -10,7 +10,7 @@
 !define VERSION_MAJOR "6"
 !define VERSION_MINOR "0"
 !define VERSION_YYYY  "2016"
-!define VERSION_MMDD  "0625"
+!define VERSION_MMDD  "0702"
 !define VERSION_PATCH ${VERSION_YYYY}${VERSION_MMDD}
 
 !define MY_ABI   "5"
Index: package/mingw-ncurses.spec
--- ncurses-6.0-20160625+/package/mingw-ncurses.spec	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/package/mingw-ncurses.spec	2016-07-02 17:00:31.000000000 +0000
@@ -3,7 +3,7 @@
 Summary: shared libraries for terminal handling
 Name: mingw32-ncurses6
 Version: 6.0
-Release: 20160625
+Release: 20160702
 License: X11
 Group: Development/Libraries
 Source: ncurses-%{version}-%{release}.tgz
Index: package/ncurses.spec
--- ncurses-6.0-20160625+/package/ncurses.spec	2016-06-25 17:17:16.000000000 +0000
+++ ncurses-6.0-20160702/package/ncurses.spec	2016-07-02 17:00:31.000000000 +0000
@@ -1,7 +1,7 @@
 Summary: shared libraries for terminal handling
 Name: ncurses6
 Version: 6.0
-Release: 20160625
+Release: 20160702
 License: X11
 Group: Development/Libraries
 Source: ncurses-%{version}-%{release}.tgz
Index: test/list_keys.c
Prereq:  1.9 
--- ncurses-6.0-20160625+/test/list_keys.c	2016-06-18 22:18:30.000000000 +0000
+++ ncurses-6.0-20160702/test/list_keys.c	2016-07-02 23:45:53.000000000 +0000
@@ -26,7 +26,7 @@
  * authorization.                                                           *
  ****************************************************************************/
 /*
- * $Id: list_keys.c,v 1.9 2016/06/18 22:18:30 tom Exp $
+ * $Id: list_keys.c,v 1.12 2016/07/02 23:45:53 tom Exp $
  *
  * Author: Thomas E Dickey
  *
@@ -88,7 +88,7 @@
 {
     int width = 0;
     char buffer[10];
-    char *value = tigetstr(name);
+    NCURSES_CONST char *value = tigetstr(name);
 
     if (show && t_opt)
 	fputc('"', stdout);
@@ -329,6 +329,7 @@
 	    printf("\n");
 	}
     }
+    free(list);
 }
 
 static void
@@ -358,7 +359,7 @@
 main(int argc, char *argv[])
 {
     int n;
-    TERMINAL **terms = typeCalloc(TERMINAL *, argc);
+    TERMINAL **terms = typeCalloc(TERMINAL *, argc + 1);
 
     while ((n = getopt(argc, argv, "ftx")) != -1) {
 	switch (n) {
@@ -383,11 +384,22 @@
     use_extended_names(x_opt);
 #endif
 
-    for (n = optind; n < argc; ++n) {
-	setupterm((NCURSES_CONST char *) argv[n], 1, (int *) 0);
-	terms[n - optind] = cur_term;
+    if (optind < argc) {
+	int found = 0;
+	int status;
+	for (n = optind; n < argc; ++n) {
+	    setupterm((NCURSES_CONST char *) argv[n], 1, &status);
+	    if (status > 0 && cur_term != 0) {
+		terms[found++] = cur_term;
+	    }
+	}
+	if (found)
+	    list_keys(terms, found);
+    } else {
+	setupterm(NULL, 1, (int *) 0);
+	terms[0] = cur_term;
+	list_keys(terms, 1);
     }
-    list_keys(terms, argc - optind);
 
     ExitProgram(EXIT_SUCCESS);
 }
